ls1intum/Artemis
# `General`: Fix slow exercise deletion in courses with many participants

### Checklist
#### General


#### Server
- [x] I adapted multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context
A couple of operation were slow due to `@Transactional` use and excessive (and sometimes even duplicated) web socket messages for edge cases, which are not really necessary. On production, I had to wait around 5min until a test exercise in a course with 2.000 participants could be deleted, when I tried to cleanup. This is completely unacceptable and another instance of really bad implementation :-(

### Description
I removed `@Transactional`, unnecessary websocket messages for edge cases and now unnecessary registration and deregistation in channels for exam users. While there might be the edge cases that a user is currently in a channel that is deleted, this is highly unlikely, because active exercises, lectures and exams are typically not deleted. In addition, we recently switched to course_wide channels for which we can implement some kind of broadcast message in such a case, which should be much faster.

In any case, this PR significantly speeds up deletion by up to 99% - it now takes a few seconds as before - and reduced the pressure on the web socket / broker subsystems by getting rid of unnecessary and sometimes even duplicated messages.

### Steps for Testing
The scenario cannot easily be tested, because newly created exercises do not add conversation_participants any more.

#### Preparation
1. Deploy an old version before #7021, e.g. release 6.4.2
2. Make sure you have a course with at least 500 participants
3. Create multiple exercises (e.g. modeling, text)
4. Delete one exercise and notice, it takes quite some time

#### Testing
1. Deploy the version of this PR
2. Delete some of the exercises created in Preparation and make sure this is done without error message and in a few seconds
3. Create an exam
4. Add user to the exam
5. Let the exam become visible
6. Make sure users can now (potentially after reloading the messages tab) see the exam channel and send messages


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
